exporter/datadogexporter/factory.go

```diff
@@ -283,11 +283,20 @@ func (f *factory) createMetricsExporter(
 		cancel()
 		return nil, fmt.Errorf("failed to start trace-agent: %w", err)
 	}
-	var statsOut chan []byte
+	acfg, err := newTraceAgentConfig(ctx, set, cfg, hostProvider)
+	if err != nil {
+		return nil, err
```

Comment:
Need to call `cancel()` before this line
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7849700829/job/21424234053?pr=31173